pipeline/builder/builder.go

```diff
@@ -168,6 +191,108 @@ func (b *Builder) buildRunJobStage(index int, s config.Stage) (*types.RunJobStag
 	return rjs, nil
 }
 
+func (b *Builder) buildV2ManifestStage(index int, s config.Stage) (*types.ManifestStage, error) {
+	ds := &types.ManifestStage{
+		StageMetadata:           buildStageMetadata(s, "deployManifest", index, b.isLinear),
+		Account:                 s.Account,
+		CloudProvider:           "kubernetes",
+		Location:                "",
+		ManifestArtifactAccount: "embedded-artifact",
+		ManifestName:            "",
+		Moniker: types.Moniker{
+			App: b.pipeline.Application,
+		},
+		Relationships: types.Relationships{LoadBalancers: []interface{}{}, SecurityGroups: []interface{}{}},
+		Source:        "text",
+	}
+
+	if len(s.Deploy.Groups) == 0 {
+		return nil, ErrNoDeployGroups
+	}
+
+	cluster := []string{s.Account, b.pipeline.Application, s.Deploy.Groups[0].Details, s.Deploy.Groups[0].Stack}
+
+	ds.Moniker.Cluster = strings.Join(cluster, "-")
+	ds.Moniker.Detail = s.Deploy.Groups[0].Details
+	ds.Moniker.Stack = s.Deploy.Groups[0].Stack
```

Comment:
This is confusing to me, there can be multiple groups, but you're popping stack and detail off of the first one. Won't this cause problems if theres 3 groups with different stacks and details?

Reply:
since a stage can have multiple manifests being applied from it, there's no way of setting the app/stack/detail of every portion.

see attached image:
![image](https://user-images.githubusercontent.com/8689924/42057429-45d20be6-7aeb-11e8-9fc8-0f33c324606f.png)

Reply:
ah, understood.